pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -413,14 +415,18 @@ class NativeAssetsBuildRunner {
         continue;
       }
       // TODO(https://github.com/dart-lang/native/issues/1321): Should dry runs be cached?
-      var (buildOutput, packageSuccess) = await _runHookForPackage(
-        hook,
-        config,
-        packageConfigUri,
-        workingDirectory,
-        includeParentEnvironment,
-        null,
-        hookKernelFile,
+      var (buildOutput, packageSuccess) =
+          await Directory.fromUri(config.outputDirectory.parent).exclusive(
```

Comment:
I think this is an overuse of extesions - I wouldn't make this an extension on `Directory`.


Isn't this much easier to read & understand:
```dart
await runUnderFileLock(..., () {
  ...
});
```
compared with this:
```dart
await Directory.fromUri(...).exclusive(..., () {
  ...
});
```
?